SpatialGDK/Documentation/content/non-unreal-layers.md

```diff
@@ -129,48 +128,31 @@ Worker_RequestId SendSomeCommandRequest(Worker_EntityId TargetEntityId, Worker_C
 #### Receive data
 You could receive and deserialize a component update and command request in your Unreal project code in the following way:
 ```
-UCLASS()
-class SPATIALGDK_API SessionComponentCallbacks : UOpCallbackTemplate
+void ATPSGameMode::BeginPlay()
 {
-    GENERATED_BODY()
+    USpatialDispatcher* Dispatcher = Cast<USpatialNetDriver>(GetWorld()->GetNetDriver())->Dispatcher;
 
-public:
-    Worker_ComponentId GetComponentId()
-    {
-        return 1337;
-    }
+    Dispatcher->AddOpCallback(1337, [&](Worker_ComponentUpdateOp Op) {
+        UWorld* World = GetWorld();
 
-    void OnComponentUpdate(const Worker_ComponentUpdateOp& Op) override
-    {
-        Worker_ComponentUpdate Update = {};
-        Update.component_id = 1338;
-        Update.schema_type = Schema_CreateComponentUpdate(1338);
-        Schema_Object* FieldsObject = Schema_GetComponentUpdateFields(Update.schema_type);
-        Schema_AddInt32(FieldsObject, 1, ++UnrealCounter);
-        Cast<USpatialNetDriver>(World->GetNetDriver())->Connection-> SendComponentUpdate(39, &Update);
-    }
+        uint32 PlayerCount = Schema_GetUint32(Schema_GetComponentUpdateFields(Op.update.schema_type), 1);
+        UE_LOG(LogTPS, Verbose, TEXT("Received update, new player count: %d"), PlayerCount);
+        const FVector Location(0.0f, 0.0f, 0.0f);
+        const FRotator Rotation = FRotator::ZeroRotator;
+        World->SpawnActor(CubeClass, &Location, &Rotation);
+    });
 
-    void OnCommandRequest(const Worker_CommandRequestOp& Op) override
-    {
+    Dispatcher->AddOpCallback(1338, [&](Worker_CommandRequestOp Op) {
         Worker_CommandResponse Response = {};
         Response.component_id = 1338;
         Response.schema_type = Schema_CreateCommandResponse(1338, 1);
-        Schema_Object* ResponseObject = Schema_GetCommandResponseObject(Response.schema_type);
+        Schema_Object* response_object = Schema_GetCommandResponseObject(Response.schema_type);
 
-        const char* Text = "My Response text.";
-        Schema_AddBytes(ResponseObject, 1, (const uint8_t*)Text, sizeof(char) * strlen(Text));
+        const char* text = "Here's my response.";
```

Comment:
nit: It might be better if this example used an FString rather than a raw char*.